src/lib/context-menu.ts

```diff
@@ -332,24 +350,48 @@ export class ContextMenu {
                           insertTableFootnote(
                             this.node,
                             this.getPos(),
+                            this.view,
                             state,
-                            dispatch
+                            dispatch,
+                            tablesFootnoteLabels
                           )
                           this.props?.popper.destroy()
                         },
                         onInsert: (notes: FootnoteNode[]) => {
                           const insertedAt = this.view.state.selection.to
+                          let inlineFootnoteIndex = 1
+                          this.view.state.doc
```

Comment:
> This will retrieve all the inline-footnotes in the document, not specifically those in the table. Perhaps consider using this.node instead.

look at the `.slice(this.getPos(), insertedAt)`  we pass the table position so will start to cut the document from table


> So, I suggest moving this logic to that function. I also recommend changing it to InlineFootnotes.filter((n) => n[1] <= insertedAt).length + 1, as it appears to be more straightforward, unless you have another reason to keep it as is.

there is no `InlineFootnotes` in this file!!

Reply:
Couldn't you take advantage of the footnotes plugin state to get the inline-footnotes you are referring to?

Reply:
there is nothing useful to give us the inserted inline footnote index